### ISSUE TITLE
PeerTube homepage markup / API has no peertube-category-list or peertube-category-embed or peertube-category-miniature functions

### ISSUE THREAD

**Issue by DalekDave** [Chocobozzz/PeerTube#6071]
### Describe the problem to be solved

In PeerTube's page markup tags/API function calls, you can insert a playlist and channel embed and miniature into home page markup, but there is no equivalent functionality for categories.

The result is that although you can pull content from one user's playlists and channels into your instance's home page, you cannot pull content from a category into your home page and display content from all your instance's community posted in a specific category or a specific list of categories.

So that makes it difficult to make your instance really community-oriented.  

### Describe the solution you would like

The PeerTube home page markup tags / API needs peertube-category-list and peertube-category-embed and peertube-category-miniature functions, so that new content from the instance's community can be automatically shown on the instance's home page.

**Comment by Chocobozzz** [comment]
Hello,

You can use `peertube-videos-list ` with `data-category-one-of` attribute no? https://docs.joinpeertube.org/api/custom-client-markup#peertube-videos-list

**Comment by DalekDave** [comment]
No, using peertube-videos-list  with data-category-one-of attribute does not fulfill the need I described originally.

**Comment by Chocobozzz** [comment]
> No, using peertube-videos-list with data-category-one-of attribute does not fulfill the need I described originally.

Then I'm sorry but I don't understand your use case. Can you explain what is lacking in `peertube-videos-list` custom component?

**Comment by DalekDave** [comment]
You cannot pull content from a category into your home page and display content from all your instance's community posted in a specific category or a specific list of categories.

The PeerTube home page markup tags / API needs peertube-category-list and peertube-category-embed and peertube-category-miniature functions, so that new content from the instance's community can be automatically shown on the instance's home page.

Sorry to repeat what I previously said, but you don't seem to have understood. Probably TL;DR

**Comment by Chocobozzz** [comment]
Unfortunately repeating the same sentences won't help me to better understand your feature request.

Please describe the UI + data displayed of each component you would like:
 * `peertube-category-list`
 * `peertube-category-embed` (there is no such 
 * `peertube-category-miniature`

For example I'm not sure to understand the purpose of `peertube-category-embed` because the embed concept in PeerTube only to display a player (that plays a playlist or a video)
